code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -552,31 +552,22 @@
 /datum/gas_reaction/bzformation/react(datum/gas_mixture/air)
 	var/list/cached_gases = air.gases
 	var/pressure = air.return_pressure()
-	// This slows down in relation to pressure, very quickly. Please don't expect it to be anything more then a snail
+	var/volume = air.return_volume()
+	var/pressure_effciency = 1/pressure
+	var/ratio_efficency = min(cached_gases[/datum/gas/nitrous_oxide][MOLES]/cached_gases[/datum/gas/plasma][MOLES],1)
+	var/bz_formed = min(pressure_effciency*ratio_efficency*volume/100, cached_gases[/datum/gas/nitrous_oxide][MOLES], cached_gases[/datum/gas/plasma][MOLES]*INVERSE(2))
```

Comment:
I would recommend keeping this trim, maybe have the volume/100 be on the pressure efficiency part and rename it to something suitable?

Reply:
> I would recommend keeping this trim, maybe have the volume/100 be on the pressure efficiency part and rename it to something suitable? We would do well to avoid the headache of how old bz is written i hope 😄 .

Will look at this tomorrow